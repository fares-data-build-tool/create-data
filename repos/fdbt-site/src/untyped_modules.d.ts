declare module 'set-cookie';

declare module 'mock-req' {
    import { IncomingMessage } from 'http';

    export default class MockReq extends IncomingMessage {
        constructor(config: object);
    }
}

declare module 'mock-res' {
    import { ServerResponse } from 'http';

    export default class MockRes extends ServerResponse {
        constructor();
    }
}

declare module 'clamscan' {
    export default class NodeClam {
        constructor();

        init: (config: object) => Promise<{ is_infected: (path: string) => { is_infected: boolean } }>;
    }
}

declare module '*.jpeg' {
    const value: string;
    export = value;
}

declare module '*.jpg' {
    const value: string;
    export = value;
}

declare module '*.png' {
    const value: string;
    export = value;
}

declare module '*.svg' {
    const value: string;
    export = value;
}

declare module '*.ico' {
    const value: {
        src: string;
    };
    export = value;
}

declare module '*.pdf' {
    const value: string;
    export = value;
}

declare module '*.csv' {
    const value: string;
    export = value;
}
