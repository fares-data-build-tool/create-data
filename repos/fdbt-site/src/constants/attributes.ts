// Session Attributes

export const FARE_TYPE_ATTRIBUTE = 'fdbt-fare-type';

export const CARNET_FARE_TYPE_ATTRIBUTE = 'fdbt-carnet-fare-type';

export const PASSENGER_TYPE_ATTRIBUTE = 'fdbt-passenger-type';

export const DEFINE_PASSENGER_TYPE_ERRORS_ATTRIBUTE = 'fdbt-passenger-type-errors';

export const SERVICE_ATTRIBUTE = 'fdbt-service';

export const JOURNEY_ATTRIBUTE = 'fdbt-journey';

export const FARE_STAGES_ATTRIBUTE = 'fdbt-fare-stages';

export const STAGE_NAMES_ATTRIBUTE = 'fdbt-stage-names';

export const DURATION_VALID_ATTRIBUTE = 'fdbt-duration-valid';

export const PRODUCT_DETAILS_ATTRIBUTE = 'fdbt-product-details';

export const TICKET_REPRESENTATION_ATTRIBUTE = 'fdbt-ticket-representation';

export const CSV_UPLOAD_ATTRIBUTE = 'fdbt-csv-upload';

export const FARE_ZONE_ATTRIBUTE = 'fdbt-csv-zone-upload';

export const PERIOD_EXPIRY_ATTRIBUTE = 'fdbt-period-expiry';

export const SERVICE_LIST_ATTRIBUTE = 'fdbt-services';

export const INPUT_METHOD_ATTRIBUTE = 'fdbt-input-method';

export const NUMBER_OF_STAGES_ATTRIBUTE = 'fdbt-number-stages';

export const MATCHING_ATTRIBUTE = 'fdbt-matching';

export const INBOUND_MATCHING_ATTRIBUTE = 'fdbt-inbound-matching';

export const NUMBER_OF_PRODUCTS_ATTRIBUTE = 'fdbt-number-of-products';

export const MULTIPLE_PRODUCT_ATTRIBUTE = 'fdbt-multiple-product';

export const PRICE_ENTRY_ATTRIBUTE = 'fdbt-price-entry';

export const SOP_ATTRIBUTE = 'fdbt-sales-offer-package';

export const SOP_INFO_ATTRIBUTE = 'fdbt-sales-offer-package-info';

export const SALES_OFFER_PACKAGES_ATTRIBUTE = 'fdbt-select-sales-offer-packages';

export const GROUP_SIZE_ATTRIBUTE = 'fdbt-group-size';

export const GROUP_PASSENGER_TYPES_ATTRIBUTE = 'fdbt-group-passenger-types';

export const GROUP_PASSENGER_INFO_ATTRIBUTE = 'fdbt-group-passenger-info';

export const GROUP_DEFINITION_ATTRIBUTE = 'fdbt-group-definition';

export const TIME_RESTRICTIONS_DEFINITION_ATTRIBUTE = 'fdbt-time-restrictions-definition';

export const RETURN_VALIDITY_ATTRIBUTE = 'fdbt-return-validity';

export const PRODUCT_DATE_ATTRIBUTE = 'fdbt-product-date-attribute';

export const MULTIPLE_OPERATOR_ATTRIBUTE = 'fdbt-multiple-operator-attribute';

export const MULTIPLE_OPERATORS_SERVICES_ATTRIBUTE = 'fdbt-multiple-operators-services-attribute';

export const FULL_TIME_RESTRICTIONS_ATTRIBUTE = 'fdbt-full-time-restrictions-attribute';

export const TERM_TIME_ATTRIBUTE = 'fdbt-term-time';

export const SCHOOL_FARE_TYPE_ATTRIBUTE = 'fdbt-school-fare-type';

export const FORGOT_PASSWORD_ATTRIBUTE = 'fdbt-reset-password';

export const USER_ATTRIBUTE = 'fdbt-user';

export const OPERATOR_ATTRIBUTE = 'fdbt-operator';

export const TXC_SOURCE_ATTRIBUTE = 'fdbt-txc-source';

export const MULTI_OP_TXC_SOURCE_ATTRIBUTE = 'fdbt-multi-op-txc-source';

export const REUSE_OPERATOR_GROUP_ATTRIBUTE = 'fdbt-reuse-operator-group';

export const SAVE_OPERATOR_GROUP_ATTRIBUTE = 'fdbt-save-operator-group';

export const SAVED_PASSENGER_GROUPS_ATTRIBUTE = 'fdbt-saved-passenger-groups';
