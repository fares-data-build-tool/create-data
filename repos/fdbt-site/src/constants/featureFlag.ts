// const dev = 'dev' === process.env.STAGE;
const devOrTest = ['test', 'dev'].includes(process.env.STAGE ?? '');
console.log(devOrTest);

export const globalSettingsDeleteEnabled = devOrTest;
export const myFaresEnabled = devOrTest;
export const saveProductsEnabled = true;
export const exportEnabled = false;
export const masterStopListEnabled = false;
