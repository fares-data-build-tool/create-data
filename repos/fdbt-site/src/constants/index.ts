export const OPERATOR_COOKIE: string = 'fdbt-operator-cookie';

export const SERVICE_COOKIE: string = 'fdbt-service-cookie';

export const FARETYPE_COOKIE: string = 'fdbt-faretype-cookie';

export const FEEDBACK_LINK: string = 'mailto:fdbt@transportforthenorth.com';

export const GOVUK_LINK: string = 'https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/';

export const CSV_INPUT_METHOD_COOKIE: string = 'fdbt-csv-input-method-cookie';

export const MANUAL_INPUT_METHOD_COOKIE: string = 'fdbt-manual-input-method-cookie';

export const INT_MAP_INPUT_METHOD_COOKIE: string = 'fdbt-interactive-map-input-method-cookie';
