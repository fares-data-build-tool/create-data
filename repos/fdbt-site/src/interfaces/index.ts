import { NextApiRequest, NextPageContext } from 'next';
import { DocumentContext } from 'next/document';
import { IncomingMessage, ServerResponse } from 'http';
import { ReactElement } from 'react';

// Session Attributes and Cookies

export interface Session {
    session: Express.Session;
}

export type NextApiRequestWithSession = NextApiRequest & Session;

export type NextPageContextWithSession = NextPageContext & {
    req: Session;
};

export type DocumentContextWithSession = DocumentContext & {
    req: Session;
};

export type IncomingMessageWithSession = IncomingMessage & Session;

export interface CookiePolicy {
    essential: boolean;
    usage: boolean;
}

export interface ProductInfo {
    productName: string;
    productPrice: string;
}

export interface ProductInfoWithErrors extends ProductInfo {
    errors: ErrorInfo[];
}

export interface DurationValidInfo {
    amount: string;
    duration: string;
    errors: ErrorInfo[];
}

export interface InputMethodInfo {
    inputMethod: string;
}

export interface Journey extends JourneyWithErrors {
    directionJourneyPattern?: string;
    inboundJourney?: string;
    outboundJourney?: string;
}

export interface JourneyWithErrors {
    errors?: ErrorInfo[];
}

export interface TicketRepresentationAttribute {
    name: string;
}

export interface TicketRepresentationAttributeWithErrors {
    errors: ErrorInfo[];
}

export interface Operator {
    name: string;
    nocCode: string;
}

export interface OperatorGroup {
    name: string;
    operators: Operator[];
}

export interface MultiOperatorInfo {
    nocCode: string;
    services: string[];
}

export interface TermTimeAttribute {
    termTime: boolean;
}

export type WithErrors<T> = {
    errors: ErrorInfo[];
} & T;

export interface UserDataUploadsProps {
    csvUploadApiRoute: string;
    csvUploadHintText: string;
    csvUploadTitle: string;
    guidanceDocDisplayName: string;
    guidanceDocAttachmentUrl: string;
    guidanceDocSize: string;
    csvTemplateDisplayName: string;
    csvTemplateAttachmentUrl: string;
    csvTemplateSize: string;
    errors: ErrorInfo[];
    detailSummary?: string;
    detailBody?: ReactElement;
    showPriceOption?: boolean;
    poundsOrPence?: string | null;
    csrfToken: string;
}

export interface GroupDefinitionWithErrors extends GroupDefinition {
    errors: ErrorInfo[];
}

export interface NumberOfStagesAttributeWithError {
    errors: ErrorInfo[];
}

export interface FareStagesAttribute {
    fareStages: string;
}

export interface FareStagesAttributeWithErrors {
    errors: ErrorInfo[];
}

export interface CsvUploadAttributeWithErrors {
    errors: ErrorInfo[];
    poundsOrPence?: string;
}

export interface GroupTicketAttribute {
    maxGroupSize: string;
}

export interface GroupTicketAttributeWithErrors extends GroupTicketAttribute {
    errors: ErrorInfo[];
}

export interface NumberOfProductsAttribute {
    numberOfProductsInput: string;
}

export interface NumberOfProductsAttributeWithErrors {
    errors: ErrorInfo[];
}

export interface MultipleProductAttribute {
    products: MultiProduct[];
}

export interface MultipleProductAttributeWithErrors extends MultipleProductAttribute {
    errors: ErrorInfo[];
}

export interface SchoolFareTypeAttribute {
    schoolFareType: string;
}

export interface MultipleOperatorsAttribute {
    selectedOperators: Operator[];
}

export interface MultipleOperatorsAttributeWithErrors extends MultipleOperatorsAttribute {
    errors: ErrorInfo[];
}

export interface ServiceListAttribute {
    selectedServices: string[];
}

export interface ServiceListAttributeWithErrors {
    errors: ErrorInfo[];
}

// Miscellaneous

export interface PassengerAttributes {
    passengerTypeDisplay: string;
    passengerTypeValue: string;
}

export interface BaseReactElement {
    id: string;
    name: string;
    label: string;
    defaultValue?: string;
    defaultChecked?: boolean;
    options?: string[];
}

export interface InputCheck {
    error: string;
    input: string;
    id: string;
}

export interface ErrorInfo {
    errorMessage: string;
    id: string;
    userInput?: string;
}

export interface Feedback {
    question: string;
    answer: string;
}

export interface PriceEntryError {
    input: string;
    locator: string;
}

export interface ResponseWithLocals extends ServerResponse {
    locals: {
        nonce: string;
        csrfToken: string;
    };
}

export interface PremadeTimeRestriction {
    name: string;
    contents: FullTimeRestriction[];
}

// AWS and Reference Data (e.g. NOC, TNDS, NaPTAN datasets)

export interface Stop {
    stopName: string;
    naptanCode: string;
    atcoCode: string;
    localityCode: string;
    localityName: string;
    parentLocalityName: string;
    qualifierName?: string;
    indicator?: string;
    street?: string;
}

export interface StopPoint {
    stopPointRef: string;
    commonName: string;
}

export interface S3NetexFile {
    name: string;
    noc: string | undefined;
    reference: string;
    fareType: string;
    productNames?: string;
    passengerType: string;
    serviceNames?: string;
    lineName?: string;
    zoneName?: string;
    sopNames: string;
    date: string;
    signedUrl: string;
    fileSize: number;
}

/* eslint-disable camelcase */
export interface CognitoIdToken {
    sub: string;
    aud: string;
    email_verified: boolean;
    event_id: string;
    'custom:noc': string;
    token_use: string;
    auth_time: number;
    iss: string;
    'cognito:username': string;
    exp: number;
    iat: number;
    email: string;
    'custom:contactable': string;
    'custom:schemeOperator': string;
    'custom:schemeRegionCode': string;
}

// Ticket Types

export type Ticket =
    | PointToPointTicket
    | GeoZoneTicket
    | PeriodMultipleServicesTicket
    | FlatFareTicket
    | SchemeOperatorTicket;

export type PointToPointTicket = SingleTicket | ReturnTicket;

export interface BaseTicket {
    nocCode: string;
    type: string;
    passengerType: string;
    ageRange?: string;
    ageRangeMin?: string;
    ageRangeMax?: string;
    proof?: string;
    proofDocuments?: string[];
    email: string;
    uuid: string;
    timeRestriction: FullTimeRestriction[];
    ticketPeriod: TicketPeriod;
}

export interface BasePointToPointTicket extends BaseTicket {
    operatorShortName: string;
    lineName: string;
    serviceDescription: string;
    products: BaseProduct[];
}

export interface SingleTicket extends BasePointToPointTicket {
    fareZones: FareZone[];
    termTime: boolean;
}

export interface ReturnTicket extends BasePointToPointTicket {
    inboundFareZones: FareZone[];
    outboundFareZones: FareZone[];
    returnPeriodValidity?: ReturnPeriodValidity;
}

export type PeriodTicket = PeriodGeoZoneTicket | PeriodMultipleServicesTicket;

export interface BasePeriodTicket extends BaseTicket {
    operatorName: string;
    products: ProductDetails[];
}

export interface PeriodGeoZoneTicket extends BasePeriodTicket {
    zoneName: string;
    stops: Stop[];
}

export interface Product {
    productName: string;
    productPrice: string;
    productDuration?: string;
    productValidity?: string;
    productDurationUnits?: string;
    productEndTime?: string;
}

export interface MultiOperatorGeoZoneTicket extends PeriodGeoZoneTicket {
    additionalNocs: string[];
}

export type GeoZoneTicket = PeriodGeoZoneTicket | MultiOperatorGeoZoneTicket;

export interface PeriodMultipleServicesTicket extends BasePeriodTicket {
    selectedServices: SelectedService[];
    termTime: boolean;
}

export interface MultiOperatorMultipleServicesTicket extends PeriodMultipleServicesTicket {
    additionalOperators: {
        nocCode: string;
        selectedServices: SelectedService[];
    }[];
}

export interface FlatFareTicket extends BaseTicket {
    operatorName: string;
    products: FlatFareProductDetails[];
    selectedServices: SelectedService[];
    termTime: boolean;
}

export interface BaseGroupTicket {
    nocCode: string;
    type: string;
    groupDefinition: GroupDefinition;
    email: string;
    uuid: string;
}

export interface SchemeOperatorTicket {
    schemeOperatorName: string;
    schemeOperatorRegionCode: string;
    nocCode?: string;
    type: string;
    passengerType: string;
    ageRange?: string;
    ageRangeMin?: string;
    ageRangeMax?: string;
    proof?: string;
    proofDocuments?: string[];
    email: string;
    uuid: string;
    timeRestriction: FullTimeRestriction[];
    ticketPeriod: TicketPeriod;
    products: ProductDetails[];
    zoneName: string;
    stops: Stop[];
    additionalNocs: string[];
}

export const isSchemeOperatorTicket = (data: Ticket): data is SchemeOperatorTicket =>
    (data as SchemeOperatorTicket).schemeOperatorName !== undefined &&
    (data as SchemeOperatorTicket).schemeOperatorRegionCode !== undefined;

// Matching Data (created by the user on the site)

export interface PassengerDetails {
    passengerType: string;
    ageRangeMin?: string;
    ageRangeMax?: string;
    proofDocuments?: string[];
    email: string;
    uuid: string;
    timeRestriction: FullTimeRestriction[];
    ticketPeriod: TicketPeriodWithInput;
    proof?: string[];
}

export interface CompanionInfo {
    passengerType: string;
    minNumber?: string;
    maxNumber: string;
    ageRangeMin?: string;
    ageRangeMax?: string;
    proofDocuments?: string[];
}

export interface GroupPassengerInfo extends CompanionInfo {
    ageRange: string;
    proof: string;
}

export interface GroupDefinition {
    maxGroupSize: number;
    companions: CompanionInfo[];
}

export interface TimeRestriction {
    startTime?: string;
    endTime?: string;
    validDays: ('monday' | 'tuesday' | 'wednesday' | 'thursday' | 'friday' | 'saturday' | 'sunday' | 'bankHoliday')[];
}

export interface ReturnPeriodValidity {
    amount: string;
    typeOfDuration: string;
}

export interface ReturnPeriodValidityWithErrors {
    amount?: string;
    typeOfDuration?: string;
    errors: ErrorInfo[];
}

export interface FareZone {
    name: string;
    stops: Stop[];
    prices: FareZonePrices[];
}

export interface FareZoneWithErrors {
    errors: ErrorInfo[];
}

export interface FareZonePrices {
    price: string;
    fareZones: string[];
}

export interface SelectedService {
    lineName: string;
    serviceCode: string;
    startDate: string;
    serviceDescription: string;
}

export interface BasicService {
    lineName: string;
    nocCode: string;
    operatorShortName: string;
    serviceDescription: string;
}

export interface SalesOfferPackageInfo {
    purchaseLocations: string[];
    paymentMethods: string[];
    ticketFormats: string[];
}

export interface SalesOfferPackageInfoWithErrors extends SalesOfferPackageInfo {
    errors: ErrorInfo[];
}

export interface SalesOfferPackage extends SalesOfferPackageInfo {
    name: string;
    description: string;
    id?: string;
}

export interface SalesOfferPackageWithErrors extends SalesOfferPackage {
    errors: ErrorInfo[];
}

export interface TicketPeriod {
    startDate?: string;
    endDate?: string;
}

export interface TicketPeriodWithInput extends TicketPeriod {
    dateInput: ProductDateInformation;
}

export interface BaseProduct {
    salesOfferPackages: SalesOfferPackage[];
}

export interface ProductWithSalesOfferPackages extends BaseProduct {
    productName: string;
}

export interface FlatFareProductDetails extends BaseProduct {
    productName: string;
    productPrice: string;
}

export interface ProductData {
    products: Product[];
}

export interface MultiOperatorInfoWithErrors {
    multiOperatorInfo: MultiOperatorInfo[];
    errors: ErrorInfo[];
}

export interface TimeBand {
    startTime: string;
    endTime: string;
}

export interface FullTimeRestriction {
    day: string;
    timeBands: TimeBand[];
}

export interface FullTimeRestrictionAttribute {
    fullTimeRestrictions: FullTimeRestriction[];
    errors: ErrorInfo[];
}

export interface TimeInput {
    timeInput: string;
    day: string;
}
export interface ProductDetails extends Product, BaseProduct {}
export interface FareStage {
    stageName: string;
    prices: {
        price: string;
        fareZones: string[];
    }[];
}

export interface UserFareStages {
    fareStages: FareStage[];
}

export interface UserFareZone {
    FareZoneName: string;
    NaptanCodes: string;
    AtcoCodes: string;
}

export interface DefinePassengerTypeWithErrors extends CompanionInfo {
    errors: ErrorInfo[];
}

export interface TimeRestrictionsDefinition extends TimeRestriction {
    timeRestrictionChoice?: string;
}

export interface TimeRestrictionsDefinitionWithErrors extends TimeRestrictionsDefinition {
    errors: ErrorInfo[];
}

export interface FareType {
    fareType: string;
}

export interface FareTypeWithErrors {
    errors: ErrorInfo[];
}

export interface GroupPassengerTypesCollection {
    passengerTypes: string[];
}
export interface GroupPassengerTypesCollectionWithErrors {
    errors: ErrorInfo[];
}

export interface MultiProduct {
    productName: string;
    productNameId: string;
    productPrice: string;
    productPriceId: string;
    productDuration: string;
    productDurationId: string;
    productDurationUnits: string;
    productDurationUnitsId: string;
    productValidity: string;
    productValidityId: string;
    productEndTime?: string;
    productEndTimeId?: string;
}

export interface MultiProductWithErrors extends MultiProduct {
    productNameError?: string;
    productPriceError?: string;
    productDurationError?: string;
    productDurationUnitsError?: string;
}

export interface PassengerType {
    passengerType: string;
    ageRange?: string;
    ageRangeMin?: string;
    ageRangeMax?: string;
    proof?: string;
    proofDocuments?: string[];
}

export interface PassengerTypeWithErrors {
    errors: ErrorInfo[];
}

export interface FaresInput {
    input: string;
    locator: string;
}

export interface FaresInformation {
    inputs: FaresInput[];
    errorInformation: PriceEntryError[];
}

export interface TicketPeriodWithErrors {
    errors: ErrorInfo[];
    dates: ProductDateInformation;
}

export interface ProductDateInformation {
    startDateDay: string;
    startDateMonth: string;
    startDateYear: string;
    endDateDay: string;
    endDateMonth: string;
    endDateYear: string;
}

export interface SelectSalesOfferPackageWithError {
    errors: ErrorInfo[];
    selected: { [key: string]: string[] };
}

export interface Service {
    service: string;
}

export interface ServiceWithErrors {
    errors: ErrorInfo[];
}

export interface ServiceType {
    lineName: string;
    startDate: string;
    description: string;
    origin?: string;
    destination?: string;
    serviceCode: string;
    dataSource?: string;
}

export interface ServicesInfo extends ServiceType {
    checked?: boolean;
}

export interface ServiceDB {
    serviceDescription: string;
    operatorShortName: string;
    journeyPatterns: JourneyPattern[];
}

export interface RawService {
    serviceDescription: string;
    operatorShortName: string;
    journeyPatterns: RawJourneyPattern[];
}

export interface JourneyPattern {
    startPoint: {
        Id: string;
        Display: string;
    };
    endPoint: {
        Id: string;
        Display: string;
    };
    stopList: string[];
}

export interface RawJourneyPattern {
    orderedStopPoints: {
        stopPointRef: string;
        commonName: string;
    }[];
}

// Components

export interface ConfirmationElement {
    name: string;
    content: string;
    href: string;
}

export interface FareTypeRadio {
    fareType: string;
    label: string;
}
export interface FareTypeRadioProps {
    standardFares: FareTypeRadio[];
    otherFares?: FareTypeRadio[];
}

export interface RadioWithoutConditionals extends BaseReactElement {
    value: string;
    radioButtonHint?: {
        id: string;
        content: string;
    };
}

export interface RadioWithConditionalInputs extends RadioWithoutConditionals {
    dataAriaControls: string;
    inputHint: {
        id: string;
        content: string;
        hidden?: boolean;
    };
    inputType: 'text' | 'checkbox' | 'date' | 'textWithUnits' | 'dropdown';
    inputs: BaseReactElement[] | PremadeTimeRestriction[];
    inputErrors: ErrorInfo[];
    selectIdentifier?: string;
}

export type RadioButton = RadioWithoutConditionals | RadioWithConditionalInputs;

export interface RadioConditionalInputFieldset {
    heading: {
        id: string;
        content: string;
        hidden?: boolean;
    };
    radios: RadioButton[];
    radioError: ErrorInfo[];
}

export interface UserAttribute {
    redirectFrom?: string;
}

export interface OperatorAttribute {
    name?: string;
    region?: string;
    nocCode?: string;
    uuid?: string;
    email?: string;
}

export interface ForgotPasswordAttribute {
    email: string;
}

export interface TxcSourceAttribute {
    source: 'tnds' | 'bods';
    hasTnds: boolean;
    hasBods: boolean;
}
