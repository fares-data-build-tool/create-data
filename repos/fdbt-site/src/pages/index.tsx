import '../design/Pages.scss';
import React, { ReactElement } from 'react';
import { NextPage } from 'next';
import Layout from '../layout/Layout';

const title = 'Fares data build tool';
const description = 'Fares data build tool is a service that allows you to generate data in NeTEx format';

const Home: NextPage = (): ReactElement => (
    <Layout title={title} description={description}>
        <main className="govuk-main-wrapper app-main-class" id="main-content" role="main">
            <h1 className="govuk-heading-xl">Fares data build tool</h1>
            <p className="govuk-body-l">Use this service to:</p>
            <div className="govuk-body-l">
                <ul className="index-page-list">
                    <li>Add fares to a newly registered bus service</li>
                    <li>Update fares on an existing bus service</li>
                    <li>Generate fares data for services in NeTEx format</li>
                </ul>
            </div>

            <a
                href="/operator"
                role="button"
                draggable="false"
                className="govuk-button govuk-button--start"
                data-module="govuk-button"
                id="start-now-button"
            >
                Start now
                <svg
                    className="govuk-button__start-icon"
                    xmlns="http://www.w3.org/2000/svg"
                    width="17.5"
                    height="19"
                    viewBox="0 0 33 40"
                    role="presentation"
                    focusable="false"
                >
                    <path fill="currentColor" d="M0 0h13l20 20-20 20H0l20-20z" />
                </svg>
            </a>
        </main>
    </Layout>
);

Home.getInitialProps = (): {} => {
    return {};
};

export default Home;
