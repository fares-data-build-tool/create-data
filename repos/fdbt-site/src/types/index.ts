export type ErrorInfo = {
    errorMessage: string;
};

export type Uuid = {
    uuid: string;
};
