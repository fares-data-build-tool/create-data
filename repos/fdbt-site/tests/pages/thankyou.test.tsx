import * as React from 'react';
import { shallow } from 'enzyme';
import { NextPageContext } from 'next';
import { mockRequest, mockResponse } from 'mock-req-res';
import { SERVICE_COOKIE } from '../../src/constants';
import Thankyou from '../../src/pages/thankyou';

describe('pages', () => {
    describe('thankyou', () => {
        it('should render correctly', () => {
            const tree = shallow(<Thankyou uuid="a1b2c3d4e5" />);
            expect(tree).toMatchSnapshot();
        });
    });
});

describe('getInitialProps', () => {
    it('retrieves the uuid correctly', () => {
        const req = mockRequest({
            body: {},
            cookies: {
                SERVICE_COOKIE:
                    '%7B%22service%22%3A%22N1%22%2C%22uuid%22%3A%22d177b8a0-44ed-4e67-9fd0-2d581b5fa91a%22%7D',
            },
            headers: {
                host: 'localhost',
                cookie: `${SERVICE_COOKIE}=%7B%22service%22%3A%22N1%22%2C%22uuid%22%3A%22d177b8a0-44ed-4e67-9fd0-2d581b5fa91a%22%7D`,
            },
        });

        const res = mockResponse({});

        const expectedResults = {
            uuid: 'd177b8a0-44ed-4e67-9fd0-2d581b5fa91a',
        };

        const props: NextPageContext = {
            pathname: '',
            query: {},
            req,
            res,
            AppTree: () => {
                return <div />;
            },
        };
        const results = Thankyou.getInitialProps(props);
        expect(results).toEqual(expectedResults);
    });
});
