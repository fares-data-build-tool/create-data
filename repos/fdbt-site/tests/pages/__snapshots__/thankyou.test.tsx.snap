// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages thankyou should render correctly 1`] = `
<Layout
  description="Thank you page for the Fares Data Build Tool"
  title="Thank You - Fares data build tool"
>
  <main
    className="govuk-main-wrapper govuk-main-wrapper--l"
    id="main-content"
    role="main"
  >
    <div
      className="govuk-grid-row"
    >
      <div
        className="govuk-panel govuk-panel--confirmation"
      >
        <h1
          className="govuk-panel__title"
        >
          Application complete
        </h1>
        <div
          className="govuk-panel__body"
          id="uuid-ref-number"
        >
          Your reference number
          <br />
          <strong>
            a1b2c3d4e5
          </strong>
        </div>
      </div>
      <h2
        className="govuk-heading-m"
      >
        What happens next
      </h2>
      <p
        className="govuk-body"
      >
        Thank you, we've submitted your fares data.
      </p>
      <p
        className="govuk-body"
      >
        We will contact you should there be any problem with the data you've provided. Your data will be converted to the NeTEx format and published.
      </p>
      <p
        className="govuk-body"
      >
        <a
          className="govuk-link"
          href="mailto:fdbt@transportforthenorth.com"
        >
          What did you think of this service?
        </a>
      </p>
      <br />
      <a
        className="govuk-button govuk-button--start"
        data-module="govuk-button"
        draggable="false"
        href="/"
        role="button"
      >
        Add another fare
      </a>
    </div>
  </main>
</Layout>
`;
